src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -230,22 +328,183 @@ public static class HwIntrinsics
+                    Vector128<byte> vmaske = Ssse3.AlignRight(vmasko, vmasko, 12).AsByte();
+
+                    Span<byte> bytes = stackalloc byte[Vector128<byte>.Count];
+                    Shuffle.MmShuffleSpan(ref bytes, control);
+                    Vector128<byte> vshuffle = Unsafe.As<byte, Vector128<byte>>(ref MemoryMarshal.GetReference(bytes));
+
+                    ref Vector128<byte> sourceBase =
+                        ref Unsafe.As<byte, Vector128<byte>>(ref MemoryMarshal.GetReference(source));
+
+                    ref Vector128<byte> destBase =
+                        ref Unsafe.As<byte, Vector128<byte>>(ref MemoryMarshal.GetReference(dest));
+
+                    int n = source.Length / Vector128<byte>.Count;
+
+                    for (int i = 0; i < n; i += 3)
+                    {
+                        ref Vector128<byte> vs = ref Unsafe.Add(ref sourceBase, i);
+
+                        Vector128<byte> v0 = vs;
+                        Vector128<byte> v1 = Unsafe.Add(ref vs, 1);
+                        Vector128<byte> v2 = Unsafe.Add(ref vs, 2);
+                        Vector128<byte> v3 = Sse2.ShiftRightLogical128BitLane(v2, 4);
+
+                        v2 = Ssse3.AlignRight(v2, v1, 8);
+                        v1 = Ssse3.AlignRight(v1, v0, 12);
+
+                        v0 = Ssse3.Shuffle(Sse2.Or(Ssse3.Shuffle(v0, vmask), vfill), vshuffle);
+                        v1 = Ssse3.Shuffle(Sse2.Or(Ssse3.Shuffle(v1, vmask), vfill), vshuffle);
+                        v2 = Ssse3.Shuffle(Sse2.Or(Ssse3.Shuffle(v2, vmask), vfill), vshuffle);
+                        v3 = Ssse3.Shuffle(Sse2.Or(Ssse3.Shuffle(v3, vmask), vfill), vshuffle);
+
+                        v0 = Ssse3.Shuffle(v0, vmaske);
+                        v1 = Ssse3.Shuffle(v1, vmasko);
+                        v2 = Ssse3.Shuffle(v2, vmaske);
+                        v3 = Ssse3.Shuffle(v3, vmasko);
+
+                        v0 = Ssse3.AlignRight(v1, v0, 4);
+                        v3 = Ssse3.AlignRight(v3, v2, 12);
+
+                        v1 = Sse2.ShiftLeftLogical128BitLane(v1, 4);
+                        v2 = Sse2.ShiftRightLogical128BitLane(v2, 4);
+
+                        v1 = Ssse3.AlignRight(v2, v1, 8);
+
+                        ref Vector128<byte> vd = ref Unsafe.Add(ref destBase, i);
+
+                        vd = v0;
+                        Unsafe.Add(ref vd, 1) = v1;
+                        Unsafe.Add(ref vd, 2) = v3;
+                    }
+                }
+            }
+
+            [MethodImpl(InliningOptions.ShortMethod)]
+            private static void Pad3Shuffle4(
+                ReadOnlySpan<byte> source,
+                Span<byte> dest,
+                byte control)
+            {
+                if (Ssse3.IsSupported)
+                {
+                    Vector128<byte> vmask = Vector128.Create(0, 1, 2, 0x80, 3, 4, 5, 0x80, 6, 7, 8, 0x80, 9, 10, 11, 0x80).AsByte();
```

Comment:
Don't know if you had planned on switching these `Vector128.Create`s to use the ROS hack, but this might be why your 96-byte benchmark case is nearly twice the time of your scalar fallback.  On netcoreapp3.1 this `Vector128.Create` compiles to:

```asm
xor         ecx, ecx
vmovd       xmm0, ecx
mov         ecx,1
vpinsrb     xmm0, xmm0, ecx,1
mov         ecx,2
vpinsrb     xmm0, xmm0, ecx,2
mov         ecx,80h
vpinsrb     xmm0, xmm0, ecx,3
mov         ecx,3
vpinsrb     xmm0, xmm0, ecx,4
mov         ecx,4
vpinsrb     xmm0, xmm0, ecx,5
mov         ecx,5
vpinsrb     xmm0, xmm0, ecx,6
mov         ecx,80h
vpinsrb     xmm0, xmm0, ecx,7
mov         ecx,6
vpinsrb     xmm0, xmm0, ecx,8
mov         ecx,7
vpinsrb     xmm0, xmm0, ecx,9
mov         ecx,8
vpinsrb     xmm0, xmm0, ecx,0Ah
mov         ecx,80h
vpinsrb     xmm0, xmm0, ecx,0Bh
mov         ecx,9
vpinsrb     xmm0, xmm0, ecx,0Ch
mov         ecx,0Ah
vpinsrb     xmm0, xmm0, ecx,0Dh
mov         ecx,0Bh
vpinsrb     xmm0, xmm0, ecx,0Eh
mov         ecx,80h
vpinsrb     xmm0, xmm0, ecx,0Fh
```

with `vpinsrb` being a 2-cycle instruction and with each of those dependent on the previous result.  The ROS hack makes it a single `vmovdqu`.

Reply:
Ah, good spot. Forgot to do that! Same elsewhere too.